src/abstractdataframe/show.jl

```diff
@@ -24,12 +24,20 @@ end
 Render a value to an IO object. Unlike
 `show`, render strings without surrounding quote marks.
 """
-ourshow(io::IO, x::Any) =
-    show(IOContext(io, :compact=>get(io, :compact, true), :typeinfo=>typeof(x)), x)
+function ourshow(io::IO, x::Any)
+    ctxt = IOContext(io, :compact=>get(io, :compact, true), :typeinfo=>typeof(x))
+    show(ctxt, "text/plain", x)
+end
+
 ourshow(io::IO, x::AbstractString) = escape_string(io, x, "")
 ourshow(io::IO, x::Symbol) = ourshow(io, string(x))
 ourshow(io::IO, x::Nothing) = nothing
 
+# https://github.com/JuliaLang/julia/pull/34730
+if VERSION < v"1.5.0-DEV.261"
```

Comment:
Just to be sure - I understand that the workaround is needed only for `AbstractChar`?

Reply:
I just double checked what was affected from the change in https://github.com/JuliaLang/julia/pull/34387 and `AbstractChar` and `Irrational` need work arounds. It's possible that other external types will also need changes.

We may want to add compact support for `Irrational` as well:

Currently with PR
```julia
julia> DataFrame(x = π)
1×1 DataFrame
│ Row │ x                      │
│     │ Irrational{:π}         │
├─────┼────────────────────────┤
│ 1   │ π = 3.1415926535897... │
```

Without PR:
```julia
julia> DataFrame(x = π)
1×1 DataFrame
│ Row │ x         │
│     │ Irration… │
├─────┼───────────┤
│ 1   │ π         │
```

Reply:
I'm fine with either option here. What are other's preferences?

Reply:
Maybe let us then also handle `Irrational` just to be less breaking (but probably this is not crucial). Other types - this is OK, let us just try to be consistent with Base.

Could you please add a test for `Char` and `Irrational` then (just as additional columns to the "dates" test)? Thank you.

Reply:
I separated the tests. If you still want me to add them to the dates test I can change it.